src/pushsource/_impl/backend/koji_source.py

```diff
@@ -134,6 +134,7 @@ def __init__(
         timeout=60 * 30,
         cache=None,
         executor=None,
+        koji_opts=None,
```

Comment:
Can we please not add this to the public API?  I think it would be fine to simply hardcode that `{"anon_retry": True}` is always used, and not allow the caller to set the options.

Nothing in the API/docs currently promises that `KojiSource` works by using a `koji.ClientSession`, it's only an implementation detail.  Adding this API is exposing that implementation detail and making it potentially harder to change in the future.